### ISSUE TITLE
Extract producer configuration with empty properties

### ISSUE THREAD

**Issue by ices1231** [smallrye/smallrye-reactive-messaging#557]
I have a problem with Producer configuration with optional empty properties. I want to configure security, but disable it in dev profile. 

**Background**
#### application.yml

```yaml
mp:
  messaging:
    outgoing:
      test-name:
        connector: smallrye-kafka
        bootstrap:
          servers: localhost:9092
        sasl:
          jaas:
            config: org.apache.kafka.common.security.scram.ScramLoginModule required username=admin password=admin;
          mechanism: SCRAM-SHA-512
        security:
          protocol: SASL_PLAINTEXT
        key:
          serializer: org.apache.kafka.common.serialization.StringSerializer
        value:
          serializer: org.apache.kafka.common.serialization.StringSerializer


"%dev":
   mp:
    messaging:
      outgoing:
      	test-name:
          sasl:
            jaas:
              config: ""
            mechanism: ""
          security:
            protocol: PLAINTEXT
```

According to https://github.com/eclipse/microprofile-config/issues/446 all is correct I got

> Caused by: java.util.NoSuchElementException: Cannot find attribute `sasl.mechanism` for channel `test-name`. Has been tried: mp.messaging.outgoing.test-name.sasl.mechanism and mp.messaging.connector.smallrye-kafka.sa
sl.mechanism
        at io.smallrye.reactive.messaging.impl.ConnectorConfig.getValue(ConnectorConfig.java:72)
        at io.smallrye.reactive.messaging.kafka.impl.JsonHelper.asJsonObject(JsonHelper.java:14)
        at io.smallrye.reactive.messaging.kafka.impl.KafkaSink.extractProducerConfiguration(KafkaSink.java:131)
        at io.smallrye.reactive.messaging.kafka.impl.KafkaSink.<init>(KafkaSink.java:40)
        at io.smallrye.reactive.messaging.kafka.KafkaConnector.getSubscriberBuilder(KafkaConnector.java:86)

**Alternative**

It is hard to make security properties optional.
Can you change [JsonHelper.java](https://github.com/smallrye/smallrye-reactive-messaging/blob/master/smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/impl/JsonHelper.java), as you did it here [default-configuration](https://github.com/smallrye/smallrye-reactive-messaging/blob/master/documentation/src/main/doc/modules/kafka/pages/default-configuration.adoc)

```java
config.getValue()  -> config.getOptionalValue()
```
Thank you in advance

**Comment by cescoffier** [comment]
That sounds reasonable.

Can you open a PR?

**Comment by cescoffier** [comment]
Closed #558
